src/main/java/core/basesyntax/strategy/impl/OperationHandlerSupply.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.strategy.OperationHandler;
+
+public class OperationHandlerSupply implements OperationHandler {
+    private StorageDao storage;
+
+    public OperationHandlerSupply() {
+        storage = new StorageDaoImpl();
+    }
+
+    @Override
+    public void valueOperation(String key, int value) {
+        storage.update(key, value);
```

Comment:
Supply, Return and Purchase have the same Overrided valueOperation method. You only update the storage. Where are the logic that realised your operation strategy: return(r) is +quantity, purchase(p) is -quantity and the others 2. What u do with quantity of fruits?